### ISSUE TITLE
Add dial option to specify that TLS should be used.

### ISSUE THREAD

**Issue by daniel-nichter** [garyburd/redigo#242]
In Dial,
```go
if do.dialTLS {
```
But t`dialTLS` isn't set in
```go
func DialTLSConfig(c *tls.Config) DialOption {
	return DialOption{func(do *dialOptions) {
		do.tlsConfig = c
	}}
}
```
Consequently, only way to get TLS enable is DialURL with `rediss`:
```go
	if u.Scheme == "rediss" {
		options = append([]DialOption{{dialTLS}}, options...)
	}
```

**Comment by garyburd** [comment]
I think it's as simple as adding function to return [dialTls](https://github.com/garyburd/redigo/blob/0d253a66e6e1349f4581d6d2b300ee434ee2da9f/redis/conn.go#L205).

**Comment by chrisvdg** [comment]
I added that DialTLSConfig and DialTLSSkipVerify call dialTls so that dialTLS is set to true in pr #278

**Comment by garyburd** [comment]
Fixed by 8bed3b5cdc78ba969d34749371f109147d9cd1b8.
